### DISCUSSION TITLE
how to do inverse dynamics in models containing extra force elements ( pathspring) in OpenSim

### DISCUSSION THREAD

**Discussion by UNNI-SANKAR** [ComputationalBiomechanicsLab/opensim-creator#912] [Q&A] [answered]
Recently i created a .osim file in opensim creator of an arm26 model containing an extra spring (mounted between the upper and lower arm) and i tried running the inverse dynamics and static optimization in Opensim to check whether the muscle activity is getting reduced or not during elbow flexion. But
1). For spring assisted arm and normal arm i am getting the same graph ( curves are coinciding). I tried changing the resting length, stiffness, dissipation values but still the curves are coinciding. I  don't understand what is the issue.
1.a). Since i had to add a spring element do i need to add external force file(spring force) or something before doing inverse dynamics, because while doing the inverse dynamics of the spring assisted arm i haven't added any extra force file. 

If i have to add an extra force file how to do that ?

**Comment by adamkewley** [comment, marked as answer]
I haven't answered this because I'm unsure what the answer is - I'm not entirely familiar with how the static optimization and inverse dynamics tools work (they're not features of OSC).

If the spring is in the model, and `appliesForce` is enabled, and it has reasonable stiffness etc. then, as far as I'm aware, it should be compatible with everything else in the ecosystem. E.g. if you forward-dynamic simulate it and monitor the effect of the spring and see something then it's fair to assume OpenSim knows about it - **but** I'm unsure whether static optimization/ID are expecting the relevant state variables to come externally.
